src/IdentityServer/Configuration/DependencyInjection/ConfigureOpenIdConnectOptions.cs

```diff
@@ -21,12 +24,26 @@ public ConfigureOpenIdConnectOptions(string[] schemes, IServiceProvider serviceP
         _serviceProvider = serviceProvider ?? throw new ArgumentNullException(nameof(serviceProvider));
     }
 
+    private static bool warnedInMemory = false;
+
     public void PostConfigure(string name, OpenIdConnectOptions options)
     {
         // no schemes means configure them all
         if (_schemes.Length == 0 || _schemes.Contains(name))
         {
             options.StateDataFormat = new DistributedCacheStateDataFormatter(_serviceProvider, name);
         }
+
+        var distributedCacheService = _serviceProvider.GetRequiredService<IDistributedCache>();
+
+        if (distributedCacheService is MemoryDistributedCache && !warnedInMemory)
```

Comment:
We could do the `!warnedInMrmory` check before resolving the cache to avoid doing that work more than once.